Sources/FluentUI_macOS/Components/MultilinePillPicker/MultilinePillPickerView.swift

```diff
@@ -18,37 +19,48 @@ public final class MultilinePillPickerView: ControlHostingView {
 	@MainActor public init(labels: [String], action: (@MainActor (Int) -> Void)? = nil) {
 		self.labels = labels
 		self.action = action
-		let picker = MultilinePillPicker(labels: labels, action: action)
-		super.init(AnyView(picker))
+		super.init(AnyView(EmptyView()))
+
+		let wrapper = MultilinePillPickerWrapper(viewModel: viewModel)
+		self.hostingView.rootView = AnyView(wrapper)
+
+		// Set up observation to keep the view model in sync.
+		bindProperty(from: self.$isEnabled, to: \.isEnabled, on: viewModel)
+		bindProperty(from: self.$labels, to: \.labels, on: viewModel)
+		bindProperty(from: self.$action, to: \.action, on: viewModel)
 	}
 
 	@MainActor required dynamic init?(coder aDecoder: NSCoder) {
-		fatalError("init(coder:) has not been implemented")
+		preconditionFailure("init(coder:) has not been implemented")
 	}
 
 	@MainActor required init(rootView: AnyView) {
-		fatalError("init(rootView:) has not been implemented")
+		preconditionFailure("init(rootView:) has not been implemented")
 	}
 
-	@MainActor public var isEnabled: Bool = true {
-		didSet {
-			updatePicker()
-		}
-	}
-	@MainActor public var labels: [String] {
-		didSet {
-			updatePicker()
-		}
-	}
-	@MainActor public var action: (@MainActor (Int) -> Void)? {
-		didSet {
-			updatePicker()
-		}
-	}
+	@MainActor @Published public var isEnabled: Bool = true
+	@MainActor @Published public var labels: [String]
+	@MainActor @Published public var action: (@MainActor (Int) -> Void)?
+
+	@MainActor private var viewModel: MultilinePillPickerViewModel = .init()
```

Comment:
I think this can be let bc the VM is not a value type?

Reply:
Yep good catch